### PR TITLE
Add support to install, cleanup and configure OCP 4.X

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds roles to install, cleanup OCP 4.X cluster on AWS.
- Adds roles to configure the cluster to be able to run
  perf and scale tests using Scale-CI pipeline.
- Adds a sample inventory to standup a cluster.
- Updates the readme with the instructions.

### CODE REVIEW

File: OCP-4.X/inventory.example [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+OPENSHIFT_INSTALL_LIBVIRT_IMAGE=
+
+# Cleanup
+OPENSHIFT_LIBVIRT_INSTALL_CLEANUP=False
+OPENSHIFT_AWS_INSTALL_CLEANUP=False
+
+# Post install
+OPENSHIFT_POST_INSTALL=True
+POLL_ATTEMPTS=600
+# Either machine.openshift.io or sigs.k8s.io
```
Comment (akrzos) [inline OCP-4.X/inventory.example]: This comment should move to right above line 67.

File: OCP-4.X/cluster.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (akrzos) [inline OCP-4.X/cluster.yml]: I think `install.yml` would be a better than for this playbook.

File: OCP-4.X/cluster.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- hosts: orchestration
+  user: "{{ ORCHESTRATION_USER | default('root') }}"
+  roles:
+    - { role: ansible-config, when: OPENSHIFT_INSTALL | default(True) | bool }
```
Comment (akrzos) [inline OCP-4.X/cluster.yml]: I am of the opinion that the ansible config should just be shipped in the directory where the playbook is intended to be executed from. In this case we include [one](https://github.com/redhat-performance/scale-ci-ansible/blob/master/ansible.cfg) in `scale-ci-ansible/`  This allows multiple ansible.cfg configurations for more flexibility, however comes at the cost of potentially executing a playbook outside it's intended directory without its intended ansible.cfg.

File: OCP-4.X/cluster.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    - { role: cleanup, when: (OPENSHIFT_AWS_INSTALL_CLEANUP | default(False) | bool) or (OPENSHIFT_LIBVIRT_INSTALL_CLEANUP | default(False) | bool) }
+    - { role: install, when: OPENSHIFT_INSTALL | default(True) | bool }
+    - { role: quickstart, when: OPENSHIFT_INSTALL | default(True) | bool }
+    - { role: post-install, when: OPENSHIFT_POST_INSTALL | default(False) | bool }
+    - { role: post-config, when: OPENSHIFT_TOOLING | default(False) | bool }
```
Comment (akrzos) [inline OCP-4.X/cluster.yml]: This is a nitpick but I am of the opinion that json nested in yaml is a bit more difficult to understand. I think simply converting the json into yaml provides a more readable playbook.

Ex:
```
- hosts: orchestration
  user: "{{ ORCHESTRATION_USER | default('root') }}"
  roles:
    - role: ansible-config
      when: OPENSHIFT_INSTALL | default(True) | bool
    - role: cleanup
      when: (OPENSHIFT_AWS_INSTALL_CLEANUP | default(False) | bool) or (OPENSHIFT_LIBVIRT_INSTALL_CLEANUP | default(False) | bool)
    - role: install
      when: OPENSHIFT_INSTALL | default(True) | bool
    - role: quickstart
      when: OPENSHIFT_INSTALL | default(True) | bool
    - role: post-install
      when: OPENSHIFT_POST_INSTALL | default(False) | bool
    - role: post-config
      when: OPENSHIFT_TOOLING | default(False) | bool
```
Let me know what you think.

File: README.md [hunk 5]
```diff
@@ -1,63 +1,40 @@
...
 
+### Run
+Clone the github repo:
 ```
-$ cp OCP-4.X/inventory.example inventory
-$ # Add orchestration host to inventory
-$ # Edit vars in OCP-4.X/vars/post-install.yml or define Environment vars (See below)
-$ time ansible-playbook -vv -i inventory OCP-4.X/post-install.yml
+$ git clone https://github.com/redhat-performance/scale-ci-ansible.git
+$ cd scale-ci-ansible/OCP-4.X
```
Comment (akrzos) [inline README.md]: In order to use the ansible.cfg, the end user would only have to cd into scale-ci-ansible and run `ansible-playbook -i inventory OCP-4.X/cluster.yml`

**Comment by chaitanyaenr** [comment]
@akrzos thanks for the review, made the changes requested. PTAL.

File: OCP-4.X/inventory.example [hunk 6]
```diff
@@ -1,2 +1,81 @@
...
 # The server that contains the kubeconfig and oc command
+
+[all:vars]
+ORCHESTRATION_USER=root
+CLUSTER_USER=core
+GOPATH="/root/.go"
+DISABLE_SELINUX=False
+
+# Install vars
+OPENSHIFT_INSTALL_USER=core
```
Comment (akrzos) [inline OCP-4.X/inventory.example]: Can we collapse `CLUSTER_USER` and `OPENSHIFT_INSTALL_USER`? Or is it possible to need/require two different values for each of these vars?

File: README.md [hunk 7]
```diff
@@ -1,63 +1,40 @@
...
+the cluster to be able to run perf and scale tests using Scale-CI pipeline. The varibles under the post-install section
+of the inventory can be modified to override the default values.
 
-PROMETHEUS_RETENTION_PERIOD
-PROMETHEUS_STORAGE_SIZE
-ALERTMANAGER_STORAGE_SIZE
+### Cleanup
+Set OPENSHIFT_INSTALL, OPENSHIFT_POST_INSTALL, OPENSHIFT_TOOLING to False and OPENSHIFT_AWS_INSTALL_CLEANUP to True in the inventory and run the playbook:
+```
+$ ansible-playbook -vv -i inventory cluster.yml
```
Comment (akrzos) [inline README.md]: This should be

```
$ ansible-playbook -vv -i inventory OCP-4.X/install.yml
```

File: OCP-4.X/roles/install/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+     dest: "{{ ansible_env.HOME }}/.aws/credentials"
+
+- name: build installer binary
+  shell: cd {{ workdir }}; hack/build.sh
+  environment:
+     GOPATH: "{{ GOPATH }}"
+
+- name: copy the config script to /tmp
+  copy:
+    src : config.py
```
Comment (akrzos) [inline OCP-4.X/roles/install/tasks/main.yml]: This file (config.py) is missing.

**Pull request merged**